x/auth/client/rest/decode.go

```diff
@@ -0,0 +1,102 @@
+package rest
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/spf13/cobra"
+	amino "github.com/tendermint/go-amino"
+)
+
+type (
+	// DecodeReq defines a tx decoding request.
+	DecodeReq struct {
+		Tx string `json:"tx"`
+	}
+
+	// DecodeResp defines a tx decoding response.
+	DecodeResp authtypes.StdTx
+)
+
+// DecodeTxRequestHandlerFn returns the decode tx REST handler. In particular,
+// it takes base64-decoded bytes, decodes it from the Amino wire protocol,
+// and responds with a json-formatted transaction.
+func DecodeTxRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req DecodeReq
+
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		err = cliCtx.Codec.UnmarshalJSON(body, &req)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		txBytes, err := base64.StdEncoding.DecodeString(req.Tx)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		var stdTx authtypes.StdTx
+		err = cliCtx.Codec.UnmarshalBinaryLengthPrefixed(txBytes, &stdTx)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		response := DecodeResp(stdTx)
+		rest.PostProcessResponse(w, cliCtx, response)
+	}
+}
+
+// txDecodeRespStr implements a simple Stringer wrapper for a decoded tx.
+type txDecodeRespTx authtypes.StdTx
+
+func (tx txDecodeRespTx) String() string {
+	return tx.String()
+}
+
+// GetDecodeCommand returns the decode command to take Amino-serialized bytes and turn it into
+// a JSONified transaction
+func GetDecodeCommand(codec *amino.Codec) *cobra.Command {
```

Comment:
why is the CLI command under `/rest`?

Reply:
Good question, I didn't see this part. Let me move it